wallet/src/de/schildbach/wallet/service/WalletFactory.kt

```diff
@@ -0,0 +1,377 @@
+package de.schildbach.wallet.service
+
+import android.net.Uri
+import android.text.format.DateUtils
+import com.google.common.base.Charsets
+import com.google.common.base.Preconditions
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.WalletApplication
+import de.schildbach.wallet.util.Crypto
+import de.schildbach.wallet.util.Io
+import de.schildbach.wallet.util.Iso8601Format
+import org.bitcoinj.core.AddressFormatException
+import org.bitcoinj.core.DumpedPrivateKey
+import org.bitcoinj.core.ECKey
+import org.bitcoinj.core.NetworkParameters
+import org.bitcoinj.script.Script
+import org.bitcoinj.wallet.DeterministicKeyChain
+import org.bitcoinj.wallet.DeterministicSeed
+import org.bitcoinj.wallet.KeyChainGroup
+import org.bitcoinj.wallet.UnreadableWalletException
+import org.bitcoinj.wallet.Wallet
+import org.bitcoinj.wallet.WalletExtension
+import org.bitcoinj.wallet.WalletProtobufSerializer
+import org.bitcoinj.wallet.authentication.AuthenticationGroupExtension
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.text.ParseException
+import java.util.LinkedList
+import javax.inject.Inject
+
+// RestoreFromFileHelper
+// WalletUtils
+// WalletApplication
+
+// WalletDataProvider (should keep track of WalletExtensions)
+
+interface WalletFactory {
+    // Onboarding
+    fun create(params: NetworkParameters): Wallet
+    fun restoreFromSeed(params: NetworkParameters, recoveryPhrase: List<String>): Wallet
+    @Throws(IOException::class)
+    fun restoreFromFile(params: NetworkParameters, backupUri: Uri, password: String): Pair<Wallet, Boolean>
+
+    // loading from persistent storage
+    fun getExtensions(params: NetworkParameters): Array<WalletExtension>
+    fun load(params: NetworkParameters, walletFile: File): Wallet
+    fun restoreFromBackup(params: NetworkParameters, backupFile: String): Wallet
```

Comment:
These two methods are not used yet, but could replace what is in WalletApplication.  

In a future state we may consider a `WalletManager` which will implement `WalletDataProvider` and move wallet related code out of WalletApplication with respect to setting the wallet object after onboarding and loading the wallet from persistent storage.